appinventor/components/src/com/google/appinventor/components/runtime/LineChart.java

```diff
@@ -29,20 +35,16 @@ public View getView() {
         return view;
     }
 
-    @SimpleProperty(category = PropertyCategory.BEHAVIOR,
-            description = "")
-    @DesignerProperty(editorType = PropertyTypeConstants.PROPERTY_TYPE_COMPONENT + ":com.google.appinventor.components.runtime.LineChartData")
-    public void ChartData(LineChartData data) {
-        // Remove this Chart from previous LineChartData component
-        if (data != null) {
-            data.removeChart(this);
-        }
+    @Override
+    public ChartModel createChartModel() {
+        // Get the View's Data object instance
+        LineData data = view.getData();
 
-        // Add Chart to LineChartData component
-        data.addChart(this);
+        // Data does not yet exist on the Chart
+        if (data == null) {
+            data = new LineData();
```

Comment:
In case there are questions about this in the future or in the present, this is to prevent crashes using the MPAndroidChart library (empty data objects have the potential to do so on certain Charts). I will add a comment about it in the source code.

Reply:
This has been refactored.